R/global_search.R

```diff
@@ -0,0 +1,52 @@
+#' GlobalSearch.search
+#'
+#' Performs Global Search based on provided filters.
+#' @param filters (optional) Query filters.
+#' @param env (optional) Custom client environment.
+#'
+#' @examples \dontrun{
+#' # No filters are applied
+#' GlobalSearch.search()
+#'
+#' # Global Beacon search
+#' GlobalSearch.search('{"entities":[["gene","BRCA2"]]}')
+#'
+#' # Type filter (only vaults)
+#' GlobalSearch.search('{"filters":[{"and":[["type__in",["vault"]]]}]}')
+#'
+#' # Advanced search
+#' GlobalSearch.search('{"filters":[{"and":[{"and":[["created_at__range",["2020-12-27","2021-12-27"]]]}]}],"query":"fuji"}')
+#'
+#' }
+#'
+#' @references
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+GlobalSearch.search <- function(filters, env = solvebio:::.solveEnv) {
```

Comment:
Add the `...` argument and then set body `body = list(...)` this will allow for additional keyword args

Reply:
Refactored and added optional parameters, so now everything that we pass into the function is key word based.

Search examples:
```R
GlobalSearch.search(query = "fuji", limit=200)
GlobalSearch.search(entities = '[["gene","BRCA2"]]')
GlobalSearch.search(entities = '[["gene","BRCA2"]]', filters = '[{"and":[{"and":[["created_at__range",["2021-11-28","2021-12-28"]]]},["type__in",["dataset"]]]}]')

```